// Copyright (c) 2010 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Sync protocol datatype extension for password data.

syntax = "proto2";

// TODO(akalin): Re-enable this once LITE_RUNTIME supports preserving
// unknown fields.

// option optimize_for = LITE_RUNTIME;

package sync_pb;

import "encryption.proto";
import "sync.proto";

// These are the properties that get serialized into the |encrypted| field of
// PasswordSpecifics.
message PasswordSpecificsData {
  optional int32 scheme = 1;
  optional string signon_realm = 2;
  optional string origin = 3;
  optional string action = 4;
  optional string username_element = 5;
  optional string username_value = 6;
  optional string password_element = 7;
  optional string password_value = 8;
  optional bool ssl_valid = 9;
  optional bool preferred = 10;
  optional int64 date_created = 11;
  optional bool blacklisted = 12;
}

// Properties of password sync objects. The actual password data is held in a
// PasswordSpecificsData that is encrypted into |encrypted|.
message PasswordSpecifics {
  optional EncryptedData encrypted = 1;
}

extend EntitySpecifics {
  optional PasswordSpecifics password = 45873;
}
